### PR TITLE
*: Fix build and readme in llvm17

### PR DESCRIPTION
**Pull request by breezewish**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflash/issues/6233

Problem Summary:

### What is changed and how it works?

```commit-message
Fix README instruction for MacOS
Fix build unit test failure in LLVM17 + MacOS
```

This doesn't work, so removed: "use CMAKE_C_COMPILER and CMAKE_CXX_COMPILER to specify the compiler"

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by Lloyd-Pottiger — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -158,20 +158,11 @@ export CXX="/usr/bin/clang++-17"
...
 ```shell
-mkdir cmake-build-debug
-cd cmake-build-debug
-
-cmake .. -GNinja -DCMAKE_BUILD_TYPE=DEBUG -DCMAKE_C_COMPILER="$(brew --prefix)/opt/llvm/bin/clang" -DCMAKE_CXX_COMPILER="$(brew --prefix)/opt/llvm/bin/clang++"
-
-ninja tiflash
+export PATH="$(brew --prefix)/opt/llvm@17/bin:$PATH"
+export CC="$(brew --prefix)/opt/llvm@17/bin/clang"
+export CXX="$(brew --prefix)/opt/llvm@17/bin/clang++"
```
Comment (JaySon-Huang) [inline README.md:165]:
```suggestion
export CXX="$(brew --prefix)/opt/llvm@17/bin/clang++"

mkdir cmake-build-debug
cd cmake-build-debug

cmake .. -GNinja -DCMAKE_BUILD_TYPE=DEBUG

ninja tiflash
```

Comment (Lloyd-Pottiger) [reply README.md:165]: still using `cmake --workflow --preset dev`

Comment (breezewish) [reply README.md]: Added as an alternative section

**Review by JaySon-Huang — APPROVED** [review]

File: dbms/src/Storages/DeltaMerge/tests/gtest_segment_bitmap.cpp [hunk 2]
```diff
@@ -125,7 +125,7 @@ class SegmentBitmapFilterTest : public SegmentTestBasic
             SegmentTestBasic::writeSegment(SEG_ID, end - begin, begin);
             if (unit.pack_size)
             {
-                db_context->getSettingsRef().set("dt_segment_stable_pack_rows", *(unit.pack_size));
+                db_context->getSettingsRef().set("dt_segment_stable_pack_rows", static_cast<UInt64>(*(unit.pack_size)));
```
Comment (gengliqi) [inline dbms/src/Storages/DeltaMerge/tests/gtest_segment_bitmap.cpp:128]:
```suggestion
                db_context->getSettingsRef().dt_segment_stable_pack_rows = *(unit.pack_size);
```

**Comment by JaySon-Huang** [comment]
happen to unstable test: https://github.com/pingcap/tiflash/issues/9662

**Pull request merged**